bigquery/google/cloud/bigquery/job.py

```diff
@@ -1949,6 +1949,15 @@ def result(self, timeout=None, retry=DEFAULT_RETRY):
         return self._client.list_rows(dest_table, selected_fields=schema,
                                       retry=retry)
 
+    def to_dataframe(self):
+        import pandas as pd
```

Comment:
I would attempt to import it at module scope, and then set a flag to indicate success.  E.g.:

```python
try:
    import pandas
except ImportError:
    _HAVE_PANDAS = False
else:
    _HAVE_PANDAS = True
```

I would then raise a ValueError inside `to_dataframe` if `_HAVE_PANDAS` is False.

Reply:
+1, you can also do:

```python
try:
    import pandas
except ImportError:
    pandas = None
```

and then check `if pandas is None`.

Also, don't alias imports in this library (I know it's very common to alias pandas and numpy elsewhere).

Reply:
Fixed in 4bc85fd